pynumdiff/kalman_smooth/_kalman_smooth.py

```diff
@@ -6,638 +6,255 @@
-    dxdt_hat = np.ravel(xhat_smooth[1,:])
-
-    if not options['backward']:
-        return x_hat, dxdt_hat
-
-    return x_hat[::-1], dxdt_hat[::-1]
-
-
-def constant_jerk(x, dt, params, options=None):
-    """
-    Run a forward-backward constant jerk RTS Kalman smoother to estimate the derivative.
-
-    :param x: array of time series to differentiate
-    :type x: np.array (float)
-
-    :param dt: time step size
-    :type dt: float
-
-    :param params: a list of two elements:
-
-                    - r: covariance of the x noise
-                    - q: covariance of the constant velocity model
-
-    :type params: list (float)
-
-
-    :param options: a dictionary indicating whether to run smoother forwards and backwards
-                    (usually achieves better estimate at end points)
-    :type params: dict {'forwardbackward': boolean}, optional
-
-    :return: a tuple consisting of:
-
-            - x_hat: estimated (smoothed) x
-            - dxdt_hat: estimated derivative of x
-
-
-    :rtype: tuple -> (np.array, np.array)
-    """
-    if len(x.shape) == 2:
-        pass
-    else:
-        x = np.reshape(x, [1, len(x)])
-
-    if options is None:
-        options = {'forwardbackward': True}
-
-    if options['forwardbackward']:
-        x_hat_f, smooth_dxdt_hat_f = __constant_jerk__(x, dt, params, options={'backward': False})
-        x_hat_b, smooth_dxdt_hat_b = __constant_jerk__(x, dt, params, options={'backward': True})
-
-        w = np.arange(0, len(x_hat_f), 1)
-        w = w/np.max(w)
-
-        x_hat = x_hat_f*w + x_hat_b*(1-w)
-        smooth_dxdt_hat = smooth_dxdt_hat_f*w + smooth_dxdt_hat_b*(1-w)
-
-        smooth_dxdt_hat_corrected = np.mean((smooth_dxdt_hat, smooth_dxdt_hat_f), axis=0)
-
-        return x_hat, smooth_dxdt_hat_corrected
-
-    return __constant_jerk__(x, dt, params, options={'backward': False})
-
-
-def known_dynamics(x, params, u=None, options=None):
-    """
-    Run a forward RTS Kalman smoother given known dynamics to estimate the derivative.
-
-    :param x: matrix of time series of (noisy) measurements
-    :type x: np.array (float)
-
-    :param params: a list of:
-                    - x0: inital condition, matrix of Nx1, N = number of states
-                    - P0: initial covariance matrix of NxN
-                    - A: dynamics matrix, NxN
-                    - B: control input matrix, NxM, M = number of measurements
-                    - C: measurement dynamics, MxN
-                    - R: covariance matrix for the measurements, MxM
-                    - Q: covariance matrix for the model, NxN
-    :type params: list (matrix)
-
-    :param u: matrix of time series of control inputs
-    :type u: np.array (float)
-
-    :param options: a dictionary indicating whether to run smoother
-    :type params: dict {'smooth': boolean}, optional
-
-    :return: matrix:
-            - xhat_smooth: smoothed estimates of the full state x
-
-    :rtype: tuple -> (np.array, np.array)
```

Comment:
The docstring indicates here and in its topline by use of the words "to estimate the derivative" that this function should be calculating a derivative, but it's not. This begs the question: Does anybody use this, since it went unnoticed? Can I break backwards compatibility for this function alone?